pkg/pillar/types/global.go

```diff
@@ -50,50 +59,45 @@ func (gs *GlobalStatus) setItemValueBool(key string, boolVal bool) {
 
 // UpdateItemValuesFromGlobalConfig - Update values of ConfigItems from
 // globalConfig
-func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc ConfigItemValueMap) {
 	// Set Int Values
-	gs.setItemValueInt("timer.config.interval", gc.ConfigInterval)
-	gs.setItemValueInt("timer.metric.interval", gc.MetricInterval)
-	gs.setItemValueInt("timer.send.timeout", gc.NetworkSendTimeout)
-	gs.setItemValueInt("timer.reboot.no.network", gc.ResetIfCloudGoneTime)
-	gs.setItemValueInt("timer.update.fallback.no.network",
-		gc.FallbackIfCloudGoneTime)
-	gs.setItemValueInt("timer.test.baseimage.update", gc.MintimeUpdateSuccess)
-	gs.setItemValueInt("timer.port.georedo", gc.NetworkGeoRedoTime)
-	gs.setItemValueInt("timer.port.georetry", gc.NetworkGeoRetryTime)
-	gs.setItemValueInt("timer.port.testduration", gc.NetworkTestDuration)
-	gs.setItemValueInt("timer.port.testinterval", gc.NetworkTestInterval)
-	gs.setItemValueInt("timer.port.timeout", gc.NetworkTestTimeout)
-	gs.setItemValueInt("timer.port.testbetterinterval", gc.NetworkTestBetterInterval)
-	gs.setItemValueInt("timer.use.config.checkpoint", gc.StaleConfigTime)
-	gs.setItemValueInt("timer.gc.download", gc.DownloadGCTime)
-	gs.setItemValueInt("timer.gc.vdisk", gc.VdiskGCTime)
-	gs.setItemValueInt("timer.gc.rkt.graceperiod", gc.RktGCGracePeriod)
-	gs.setItemValueInt("timer.download.retry", gc.DownloadRetryTime)
-	gs.setItemValueInt("timer.boot.retry", gc.DomainBootRetryTime)
-	gs.setItemValueInt("storage.dom0.disk.minusage.percent",
-		gc.Dom0MinDiskUsagePercent)
+	gs.setItemValueInt("timer.config.interval", gc.GetGlobalValue(ConfigInterval).IntValue)
```

Comment:
GetGlobalValue(ConfigInterval).IntValue
  --> use gc.GlobalConfigItemIntValue(ConfigInterval)

Repeat for the rest of places..